### ISSUE TITLE
Application hangs on Scroll Navigation View in WinUI UWP after upgrade of WinUI 3 preview 4

### ISSUE THREAD

**Issue by Berkunath** [microsoft/microsoft-ui-xaml#4204]
I have created application with NavigationView with MenuItemsSource. Navigation View have scrollviewer due to insufficient space. While scrolling NavigationView items, application get hangs. The same issue occured in WinUI 3 Preview 2 in WinUI Desktop. After upgrade to WinUI 3 Preview 4, issue fixed in WinUI Desktop [3598](https://github.com/microsoft/microsoft-ui-xaml/issues/3598) . But the same occurred in WinUI UWP.

**Steps to reproduce the bug**
Run the attached WinUI UWP application 
Scroll the Navigation view, application get hangs
If application not hangs on initial scroll, it works smoothly
If application not hangs stop the application and try the steps from step 1

**Expected behavior**
Application should not hang on scroll navigation view

NuGet package version: 

[Microsoft.WinUI 3.0.0-preview4.210210.4] 

Windows Version:

Version 1909 (OS Build 18363.1316) reproduced frequently
Version 20H2(OS Build 19042.804) Tried multiple times but could not reproduce
Sample : 
[Scroll Navigation UWP.zip](https://github.com/microsoft/microsoft-ui-xaml/files/5993283/Scroll.Navigation.UWP.zip)

**Comment by Berkunath** [comment]
The same issue can be reproduced in ListView also.

**Comment by StephenLPeters** [comment]
@ojhad FYI

**Comment by syncsiva** [comment]
Hi All, any update on this issue?

**Comment by codendone** [comment]
Is there anything special about the repro? I tried the sample in the original post, and I'm currently able to repro. I tried scrolling with the scrollbar, with mouse wheel, and with touch. I tried with the NavigationView in normal, compact, and overlay mode. I tried on a recent OS build and 1903. What am I missing?

If you break into the app when it is hung, what is the callstack for the UI thread?

**Comment by syncsiva** [comment]
@codendone I have replicated this issue in simple app.

**Steps to reproduce the bug**
1. Run the sample.
2. Click the button
3. Scroll or mouse wheel over navigation view where app hangs.

Looks like some problem with re-using of the items. 
[project reunion uwp (2).zip](https://github.com/microsoft/microsoft-ui-xaml/files/6276167/project.reunion.uwp.2.zip)

**Version Info**
Microsoft.ProjectReunion -> Version="0.5.0-prerelease
Visual Studio : VS 2019 16.10 Preview 1.0
Microsoft Windows Version 20H2 (OS build 19042.867)

**Additional Info**
I was able to workaround this issue with different ways. If I have same ItemsSource which i'm setting to NavigationView in button click while loading itself, then i'm not encountering this issue.
